vsdeinterlace/funcs.py

```diff
@@ -15,141 +15,97 @@
-                elif invpos > 1:
-                    pos = [2, 6]
-                    assumefps = 1
-            else:
-                cleanpos = 1
-                pos = [3, 4]
-
-                if invpos > 1:
-                    cleanpos = 6
-                    assumefps = 1
-
-                if invpos > 3:
-                    pos = [4, 8]
-                    assumefps = 1
-
-            clean = bobbed_clip.std.SelectEvery(cycle, [cleanpos - invpos])
-            jitter = bobbed_clip.std.SelectEvery(cycle, [p - invpos for p in pos])
-
-            if assumefps:
-                jitter = core.std.AssumeFPS(
-                    bobbed_clip[0] * assumefps + jitter, **(ivtc_fps_div if cleanpos == 6 else ivtc_fps)
-                )
-
-            mv = MVTools(jitter, **mv_args)
-            mv.analyze()
-            comp = mv.flow_interpolate(interleave=False)
-
-            out = intl([*comp, clean], decimate, [0])
-            offs = 3 if decimate else 2
-
-            return out[invpos // offs:]
-
-        if pattern == 0:
-            c1pos = [0, 2, 7, 5]
-            c2pos = [3, 4, 9, 8]
-
-            if offset == -1:
-                c1pos = [2, 7, 5, 10]
-
-            if offset == 1:
-                c2pos = []
-                c2 = core.std.Interleave([bb(4), bb(5), bb(0, True), bb(9)])
+    def __call__(
+        self,
+        clip: vs.VideoNode,
+        pattern: int,
+        preset: MVToolsPreset = MVToolsPreset.HQ_COHERENCE,
+        blksize: int | tuple[int, int] = 16,
+        refine: int = 1,
+        thsad_recalc: int | None = None,
+    ) -> vs.VideoNode:
+        def select_every(clip: vs.VideoNode, cycle: int, offsets: int | list[int]) -> vs.VideoNode:
+            def shift_clip(clip: vs.VideoNode, n: int) -> vs.VideoNode:
+                if n > 0:
+                    return clip[n:] + clip[-1] * n
+                elif n < 0:
+                    return clip[0] * -n + clip[:n]
+                else:
+                    return clip
+
+            def select_clip(clip: vs.VideoNode, cycle: int, offsets: list[int]) -> list[vs.VideoNode]:
+                clips = list[vs.VideoNode]()
+                for x in offsets:
+                    shifted = shift_clip(clip, x)
+                    if cycle != 1:
+                        shifted = shifted.std.SelectEvery(cycle, 0)
+                    clips.append(shifted)
+                return clips
+            
+            if isinstance(offsets, int):
+                offsets = [offsets]
+
+            return core.std.Interleave(select_clip(clip, cycle, offsets))
+
+        def _floor_div_tuple(x: tuple[int, int]) -> tuple[int, int]:
+            return (x[0] // 2, x[1] // 2)
+
+        assert check_variable(clip, TelopResample)
+
+        InvalidFramerateError.check(TelopResample, clip, (60000, 1001))
+
+        mod = 10 if self == TelopResample.IVTC_TXT30 else 5
+        field_ref = pattern * 2 % mod
+        invpos = mod - field_ref % mod
```

Comment:
Is this intended to be `(mod - field_ref) % mod`? Otherwise, this `% mod` is a no-op.